pyo3-ffi/src/dictobject.rs

```diff
@@ -96,6 +104,24 @@ pub unsafe fn PyDictViewSet_Check(op: *mut PyObject) -> c_int {
     (PyDictKeys_Check(op) != 0 || PyDictItems_Check(op) != 0) as c_int
 }
 
+#[cfg(not(Py_3_13))]
+pub unsafe fn PyDict_GetItemRef(
+    dp: *mut PyObject,
+    key: *mut PyObject,
+    result: *mut *mut PyObject,
+) -> c_int {
+    let item: *mut PyObject = PyDict_GetItemWithError(dp, key);
+    if !item.is_null() {
+        *result = _Py_NewRef(item);
+        return 1; // found
+    }
+    *result = std::ptr::null_mut();
+    if !PyErr_Occurred().is_null() {
```

Comment:
This is wrong, there shouldn't be any negation: `PyErr_Occurred()` is `NULL` if an error *hasn't* occurred.